android/src/ti/imagefactory/ImageFactoryModule.java

```diff
@@ -148,89 +162,142 @@ private TiBlob imageTransform(int type, TiBlob blob, KrollDict args)
 		image.recycle();
 		image = null;
 		ref = null;
-		
+
 		return result;
 	}
 
- 	// Public Image Methods
+	@Kroll.method
+	public TiBlob fixOrientation(TiBlob blob) {
+
+		InputStream inputStream = null;
+
+		try {
+
+			inputStream = blob.getInputStream();
+			ExifInterface Exif = new ExifInterface(inputStream);
+
+			int orientation = Exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL);
+
+			KrollDict dict = new KrollDict();
+
+			switch (orientation) {
+			case ExifInterface.ORIENTATION_FLIP_HORIZONTAL:
+				dict.put("flipHorizontal", true);
+				break;
+			case ExifInterface.ORIENTATION_FLIP_VERTICAL:
+				dict.put("degrees", 180);
+				dict.put("flipHorizontal", true);
+				break;
+			case ExifInterface.ORIENTATION_TRANSPOSE:
+				dict.put("degrees", 270);
+				dict.put("flipHorizontal", true);
+				break;
+			case ExifInterface.ORIENTATION_TRANSVERSE:
+				dict.put("degrees", 90);
+				dict.put("flipHorizontal", true);
+				break;
+			case ExifInterface.ORIENTATION_ROTATE_90:
+				dict.put("degrees", 90);
+				break;
+			case ExifInterface.ORIENTATION_ROTATE_180:
+				dict.put("degrees", 180);
+				break;
+			case ExifInterface.ORIENTATION_ROTATE_270:
+				dict.put("degrees", 270);
+				break;
+			}
+
+			blob = imageTransform(TRANSFORM_ROTATE, blob, dict);
```

Comment:
Before calling `imageTransform()` we should check if the `dict` is empty. The idea is we do **NOT** want to create a new blob if the image is already upright. Meaning that this module method should return the given blob argument if it's already upright (which will always be the case for a PNG). This will greatly improve performance.

So, something like this...
```javascript
if (!dict.isEmpty()) {
	blob = imageTransform(TRANSFORM_ROTATE, blob, dict);
}
```

Reply:
Side Note:
I know that iOS always decodes a JPEG as upright for us. So, if we were to implement this `imageAsUpright()` method on iOS too, then it would always return the given blob and act as a simple pass-through.